### PR TITLE
syncer: ignore TRUNCATE TABLE in optimistic mode

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
ignore TRUNCATE TABLE in optimistic mode

### What is changed and how it works?
filter out this statement in pre-check part

### Check List 

Tests 

 - Integration test

Code changes

Side effects

Related changes

### CODE REVIEW

File: syncer/syncer.go [hunk 1]
```diff
@@ -1800,6 +1801,13 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
 				}
 			}
 		}
+		if s.cfg.ShardMode == config.ShardOptimistic {
```
Comment (csuzhangxc) [inline syncer/syncer.go:1804]: How about changing `if` to `else if`? 😂

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM

File: syncer/syncer.go (hunk 1 above)
Comment (lance6716) [reply syncer/syncer.go]: doen in https://github.com/pingcap/dm/pull/905/commits/df32d37e8f3e8f0ca08f4fbdc7e62c39a4fc6c53

**Review by csuzhangxc — COMMENTED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**